kvm-libvirt/flatcar-linux/kubernetes/controllers.tf

```diff
@@ -0,0 +1,101 @@
+resource "random_string" "volumepath" {
+  length = 6
+  special = false
+}
+
+resource "libvirt_pool" "volumetmp" {
+  name = "vms${random_string.volumepath.result}"
+  type = "dir"
+  path = "/var/tmp/vms${random_string.volumepath.result}"
+}
+
+resource "libvirt_volume" "base" {
+  name   = "${var.cluster_name}-base"
+  source = "${var.os_image_unpacked}"
+  pool   = "${libvirt_pool.volumetmp.name}"
+  format = "qcow2"
+}
+
+resource "libvirt_volume" "controller-disk" {
+  name             = "${var.cluster_name}-controller-${count.index}.qcow2"
+  count            = "${var.controller_count}"
+  base_volume_id   = "${libvirt_volume.base.id}"
+  pool             = "${libvirt_pool.volumetmp.name}"
+  format           = "qcow2"
+}
+
+
+resource "libvirt_ignition" "ignition" {
+  name    = "${var.cluster_name}-controller-${count.index}-ignition"
+  pool    = "${libvirt_pool.volumetmp.name}"
+  count   = "${var.controller_count}"
+  content = "${element(data.ct_config.controller-ignitions.*.rendered, count.index)}"
+}
+
+resource "libvirt_domain" "controller-machine" {
+  count  = "${var.controller_count}"
+  name   = "${var.cluster_name}-controller-${count.index}"
+  vcpu   = "1"
+  memory = "2048"
+
+  coreos_ignition = "${element(libvirt_ignition.ignition.*.id, count.index)}"
+
+  disk {
+    volume_id = "${element(libvirt_volume.controller-disk.*.id, count.index)}"
+  }
+
+  graphics {
```

Comment:
Haven't tried yet!

Reply:
I haven't tried it still because it's actually nice if the VM has a working console where it shows that it has booted and some other basic information even though you could login with ssh (assuming nothing broke the boot).

I also read there was a bug when it's disabled, but not sure.

Reply:
Fair enough :)